tests/delete/tests.py

```diff
@@ -436,6 +437,36 @@ def test_proxied_model_duplicate_queries(self):
         with self.assertNumQueries(2):
             avatar.delete()
 
+    def test_only_referenced_fields_selected(self):
+        """
+        Only referenced fields are selected during cascade deletion SELECT
+        unless deletion signals are connected.
+        """
+        origin = Origin.objects.create()
+        expected_sql = str(
+            Referrer.objects.only(
+                # Both fields are referenced by SecondReferrer.
+                'id', 'unique_field'
+            ).filter(origin__in=[origin]).query
+        )
+        with self.assertNumQueries(2) as ctx:
+            origin.delete()
+        self.assertEqual(ctx.captured_queries[0]['sql'], expected_sql)
+        # But all fields are selected if deletion signals are connected.
+        signals = ['pre_delete', 'post_delete']
+
+        def receiver(instance, **kwargs):
+            pass
+        for signal_name in signals:
+            with self.subTest(signal=signal_name):
+                origin = Origin.objects.create()
+                signal = getattr(models.signals, signal_name)
+                signal.connect(receiver, sender=Referrer)
+                with self.assertNumQueries(2) as ctx:
+                    origin.delete()
+                self.assertIn('large_field', ctx.captured_queries[0]['sql'])
```

Comment:
We should convert column name i.e. `connection.introspection.identifier_converter('large_field')`.